### ISSUE TITLE
Error due to deprecated Python when using Home Assistant 2023.6.0

### ISSUE THREAD

**Issue by mcdhome** [alandtse/tesla#627]
The latest release of Home Assistant will throw loading errors for Tesla Custom.

```
AttributeError: module 'asyncio' has no attribute 'coroutine'
```

Found this page explaining.  [StackOverflow LINK](https://stackoverflow.com/questions/74345065/attributeerror-module-asyncio-has-no-attribute-coroutine-in-python-3-11)
Summary "Generator-based Coroutines which contains @asyncio.coroutine decorator is removed since Python 3.11 so asyncio module doesn't have @asyncio.coroutine decorator as the error says"

I had to make edits in the Tesla Gateway \_\_init\_\_.py in custom_integrations folder.

I commented out the three lines with 
```
#@asyncio.coroutine
```

Then I had to edit this line and add 'async ' at the front.  Then the Tesla Custom loaded fine on Home Assistant restart with no errors.
```
async def async_setup(hass, config):
```

**Comment by alandtse** [comment]
Thanks. Happy to take a PR to fix this.

**Comment by zipzagster** [comment]
II have the following the the log, which I assume is related:

2023-06-08 16:39:47.152 WARNING (MainThread) [homeassistant.helpers.entity] Entity switch.skippy_polling (<class 'custom_components.tesla_custom.switch.TeslaCarPolling'>) is using self.async_update_ha_state(), without enabling force_update. Instead it should use self.async_write_ha_state(), please report it to the custom integration author.

**Comment by iscraigh** [comment]
> The latest release of Home Assistant will throw loading errors for Tesla Custom.
> 
> ```
> AttributeError: module 'asyncio' has no attribute 'coroutine'
> ```
> 
> Found this page explaining. [StackOverflow LINK](https://stackoverflow.com/questions/74345065/attributeerror-module-asyncio-has-no-attribute-coroutine-in-python-3-11) Summary "Generator-based Coroutines which contains @asyncio.coroutine decorator is removed since Python 3.11 so asyncio module doesn't have @asyncio.coroutine decorator as the error says"
> 
> I had to make edits in the Tesla Gateway __init__.py in custom_integrations folder.
> 
> I commented out the three lines with
> 
> ```
> #@asyncio.coroutine
> ```
> 
> Then I had to edit this line and add 'async ' at the front. Then the Tesla Custom loaded fine on Home Assistant restart with no errors.
> 
> ```
> async def async_setup(hass, config):
> ```

I am unable to locate the referenced folder. I have custom components and a tesla folder but nothing with that filename. I don't see a custom integration folder.

Thanks
Craig

**Comment by purcell-lab** [comment]
> 2023-06-08 16:39:47.152 WARNING (MainThread) [homeassistant.helpers.entity] Entity switch.skippy_polling (<class 'custom_components.tesla_custom.switch.TeslaCarPolling'>) is using self.async_update_ha_state(), without enabling force_update. Instead it should use self.async_write_ha_state(), please report it to the custom integration author.

The async_update issue seems to be fixed by #606 , the broader issue running under Python 3.11 maybe still outstanding although I can't reproduce on my setup.
